django/db/migrations/state.py

```diff
@@ -91,7 +91,10 @@ class ProjectState:
     def __init__(self, models=None, real_apps=None):
         self.models = models or {}
         # Apps to include from main registry, usually unmigrated ones
-        self.real_apps = real_apps or []
+        if real_apps:
+            self.real_apps = real_apps if isinstance(real_apps, set) else set(real_apps)
```

Comment:
Unless this is for performance (and if so, fair enough), just unconditionally doing `set(real_apps)` would be fine even if `real_apps` is already a set, I think (Unless you're looking to keep a reference to the _original_ set passed in?)

Reply:
ticket-29898 is to optimize schema editors, so I'd leave `isinstance`, it's ~10% faster, and internally it's always a `set`.

Reply:
Yeah I'd absolutely leave it as `isinstance` too if performance is the reason. It's a constant time cost, and as you say, is faster (pretty sure it's _more_ than 10% faster if the incoming `set` is large enough, YMMV) 👍

Reply:
> and internally it's always a set.

Are `ProjectState` objects always created internally? Is there any reason you couldn't change things so this could be `assert isinstance(real_apps, set)`?